pkg/types/openstack/validation/realvalidvaluesfetcher.go

```diff
@@ -194,3 +197,47 @@ func (f realValidValuesFetcher) GetServiceCatalog(cloud string) ([]string, error
 
 	return serviceCatalogNames, nil
 }
+
+func (f realValidValuesFetcher) GetFloatingIPNames(cloud string, floatingNetworkName string) ([]string, error) {
+	opts := &clientconfig.ClientOpts{
+		Cloud: cloud,
+	}
+
+	conn, err := clientconfig.NewServiceClient("network", opts)
+	if err != nil {
+		return nil, err
+	}
+
+	// floatingips.ListOpets requires an ID so we must get it from the name
+	floatingNetworkID, err := networks.IDFromName(conn, floatingNetworkName)
+	if err != nil {
+		panic(err)
```

Comment:
Please don't panic. I think it's better if you just return the error

Reply:
Oh of course, sorry! This was an oversight I forgot to fix.